sdk/src/types/block/address/bech32.rs

```diff
@@ -18,67 +18,40 @@ use packable::{
 
 use crate::types::block::{address::Address, ConvertTo, Error};
 
-const HRP_MAX: u8 = 83;
-
-#[derive(Copy, Clone, Eq, PartialEq, Ord, PartialOrd, Hash)]
-pub struct Hrp {
-    inner: [u8; HRP_MAX as usize],
-    len: u8,
-}
+#[derive(Copy, Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Deref)]
+#[repr(transparent)]
+pub struct Hrp(bech32::Hrp);
 
 impl core::fmt::Debug for Hrp {
     fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
         f.debug_struct("Hrp")
-            .field("display", &self.to_string())
-            .field("inner", &prefix_hex::encode(&self.inner[..self.len as usize]))
-            .field("len", &self.len)
+            .field("display", &self.0.to_string())
+            .field("bytes", &prefix_hex::encode(self.0.byte_iter().collect::<Vec<_>>()))
+            .field("len", &self.0.len())
```

Comment:
we can remove len as well now i think

Reply:
Can't we just use the crate's Debug impl actually ? What does it look like?

Reply:
Let's keep ours, this is not helpful